src/server/KbnServer.js

```diff
@@ -73,6 +73,7 @@ module.exports = class KbnServer {
     let { server, config } = this;
 
     await this.ready();
+    server.listener.timeout = config.get('elasticsearch.requestTimeout');
```

Comment:
First issue is that this is using the elasticsearch plugins's config in `KbnServer`. Ideally, this would be something that happens at the route level, no? The elasticsearch plugin could define the timeout for the proxy route to match it's `requestTimeout` configuration.